app/models/html_attachment.rb

```diff
@@ -66,7 +66,7 @@ def url(options = {})
       options[:host] = URI(Plek.new.external_url_for("www-origin")).host
     end
 
-    type = attachable.class.name.underscore
+    type = attachable.try(:html_attachment_type) || attachable.class.name.underscore
```

Comment:
I'm just thinking about how we ended up in a situation where the path to the HTML attachment didn't match the parent document and I think it's because the code on this line ends up "out of sync" with this code here:

https://github.com/alphagov/whitehall/blob/5d76209841f3e25ebe2c7c8d0b944ce934a12c80/app/helpers/public_document_routes_helper.rb#L32-L36

I was wondering if there was a way we could have both of these call a shared function so the logic always stays the same.

Reply:
Thanks for this, I hadn't realised that. I have added a `base_path` method to the classes with HTML Attachments. As ConsultationOutcomes also allow HTML Attachments but are not editions it's been added to a few places but hopefully breaks the reliance and places the knowledge back in the owning class.

Though I'm not sure of the method name.